### PR TITLE
Add ability to make requests to OpenAI api

### PR DESCRIPTION
**Pull request by Tom0Brien**
This PR adds a module that integrates with OpenAI's api to send text prompts and receive generated responses. Additionally, the responses can be "spoken" using the "Say" skill. I am not sure whether or not it makes sense to integrate this within the Director system or not.

### CODE REVIEW

File: module/skill/GPT/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Include this module to allow your system to interact with OpenAI's GPT for natural language processing tasks.
+
+## Consumes
+
+- `message::skill::GPTRequest`: A request containing text that needs to be processed by GPT.
+
+## Emits
+
+- `message::skill::GPTResponse`: A response from GPT containing the generated text based on the input request.
```
Comment (ysims) [inline module/skill/GPT/README.md:17]:
```suggestion
- `message::skill::Say`: A Task requesting to speak using the output of the GPT model based on the input request.
```

**Review by ysims — COMMENTED** [review]
Same comment about the tester here from the speech PR
The GPT module can have a test config value to use for simpler testing. 
A role with just Say and GPT wuld be useful. then leave `behaviour.role` for soccer behaviour testing.

File: module/skill/GPT/data/config/GPT.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Controls the minimum log level that NUClear log will display
+log_level: INFO
+
+# The OpenAI API costs $$$ to use, so we need to use a key to access it. DO NOT COMMIT THIS
+openai_api_key: ""
```
Comment (jpptm) [inline module/skill/GPT/data/config/GPT.yaml:5]:
```suggestion
# The OpenAI API costs $$$ to use, so we need to use a key to access it. This must be filled before compiling. DO NOT COMMIT THE KEY IN THE REPOSITORY.
openai_api_key: ""
```

File: module/skill/GPT/src/GPT.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        });
+
+        on<Provide<GPTRequest>>().then([this](const GPTRequest& gpt_request, const RunInfo& info) {
+            if (info.run_reason == RunInfo::NEW_TASK) {
+                // Send request to OpenAI API
+                nlohmann::json request = {
+                    {"model", "gpt-3.5-turbo"},
+                    {"messages", nlohmann::json::array({{{"role", "user"}, {"content", gpt_request.text}}})},
+                    {"max_tokens", 100},
+                    {"temperature", 0}};
```
Comment (jpptm) [inline module/skill/GPT/src/GPT.cpp:36]: Should these be configurable values?

**Review by jpptm — CHANGES_REQUESTED** [review]
Nitpicking

File: module/skill/GPT/src/GPT.cpp (hunk 3 above)
Comment (Tom0Brien) [reply module/skill/GPT/src/GPT.cpp:36]: Yes, good point.

**Pull request closed without merging**